plugin/src/test/kotlin/org/jlleitschuh/gradle/ktlint/testdsl/testAnnotations.kt

```diff
@@ -1,21 +1,28 @@
 package org.jlleitschuh.gradle.ktlint.testdsl
 
+import java.io.File
+import java.util.stream.Stream
+import kotlin.streams.asStream
 import org.gradle.util.GradleVersion
 import org.jetbrains.kotlin.utils.addToStdlib.cast
 import org.jlleitschuh.gradle.ktlint.KtlintBasePlugin
 import org.junit.jupiter.api.extension.ExtensionContext
 import org.junit.jupiter.params.provider.Arguments
 import org.junit.jupiter.params.provider.ArgumentsProvider
```

Comment:
This currently is failing CI due to ```
/home/runner/work/ktlint-gradle/ktlint-gradle/plugin/src/test/kotlin/org/jlleitschuh/gradle/ktlint/testdsl/testAnnotations.kt:3:1 Imports must be ordered in lexicographic order without any empty lines in-between with "java", "javax", "kotlin" and aliases in the end
```
But, unless I forgot my ABCs, this already is in lexicographic order, and wasn't before!

Reply:
Looks like "java" stuff should go **after** other things.
See example here: https://github.com/JLLeitschuh/ktlint-gradle/blob/main/plugin/src/test/kotlin/org/jlleitschuh/gradle/ktlint/AbstractPluginTest.kt

```kotlin
package org.jlleitschuh.gradle.ktlint

import org.jlleitschuh.gradle.ktlint.tasks.GenerateReportsTask
import org.jlleitschuh.gradle.ktlint.testdsl.GradleTestVersions
import org.junit.jupiter.api.io.TempDir
import java.io.File

@GradleTestVersions
abstract class AbstractPluginTest {
...
```

Reply:
not sure yet where "kotlin" should go, but will try to find an example.

Reply:
Found it! Example: https://github.com/JLLeitschuh/ktlint-gradle/blob/main/plugin/src/test/kotlin/org/jlleitschuh/gradle/ktlint/TaskConfigurationAvoidanceTest.kt

```kotlin
package org.jlleitschuh.gradle.ktlint

import org.gradle.util.GradleVersion
import org.jlleitschuh.gradle.ktlint.tasks.GenerateReportsTask
import org.jlleitschuh.gradle.ktlint.tasks.KtLintCheckTask
import org.jlleitschuh.gradle.ktlint.tasks.KtLintFormatTask
import org.jlleitschuh.gradle.ktlint.tasks.LoadReportersTask
import org.jlleitschuh.gradle.ktlint.testdsl.GradleArgumentsProvider
import org.jlleitschuh.gradle.ktlint.testdsl.GradleTestVersions
import org.jlleitschuh.gradle.ktlint.testdsl.build
import org.jlleitschuh.gradle.ktlint.testdsl.project
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.extension.ExtensionContext
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.ArgumentsSource
import java.util.stream.Stream
import kotlin.streams.asStream

@GradleTestVersions
class TaskConfigurationAvoidanceTest : AbstractPluginTest() {
```

So Java & Kotlin should go separately **after** third-party stuff, without empty lines between third-party stuff and  Java & Kotlin stuff, but themselves, they must be ordered lexicographically. That's it.

Reply:
Anyone can fix this?

Reply:
@tommyettinger Of course, this is not an issue, but we should treat it as _something_ that violates the established rules of static analysis. This is the usual code hygiene, keeping the same style. I believe that it is better to politely ask the contributor to update the pull request than to appeal to the repository owner to merge anything just because we want it asap. :)

Reply:
> That explains it, I suppose, but doesn't really tell us why it's a build-stopping error... Doesn't IntelliJ IDEA reorder imports on its own rather often?

There are multiple approaches to code formatting in the project. If you don't care - then you don't care. If you care a bit, you use some linting tool, but allow its failures. If you strongly care - any formatting issue must be fixed, and that's what we see here. And this is not something unusual. Dor example, in my project we also use ktlint, and ktlint issue does fail the build too, because **we do care** about formatting of our code to that extent. So finally, to say true, I'm not interested in discussing too much the philosophy, whether it is right or wrong, it's up to project maintainers to define that. I'm most interested to have this merged and relased ASAP with whatever fixes must be done here to make that happen.

Reply:
And yes, default formatting in the IntelliJ may be diffrent from what is configured for ktlint, so at least you must have ktlint plugin for IntelliJ installed to be able to reformat accordingly or run something like `./gradlew ktlintFormat` in your pre-commit hook (but I would recommend to install the plugin).

Reply:
Sorry about that! I've updated this.